packages/lesswrong/lib/collections/tags/helpers.ts

```diff
@@ -54,7 +56,7 @@ export const tagGetCommentLink = ({tagSlug, commentId, tagCommentType = "DISCUSS
   isAbsolute?: boolean,
 }): string => {
   const base = tagCommentType === "DISCUSSION" ? tagGetDiscussionUrl({slug: tagSlug}, isAbsolute) : tagGetSubforumUrl({slug: tagSlug}, isAbsolute)
-  return commentId ? `${base}#${commentId}` : base
+  return commentId ? `${base}?commentId=${commentId}` : base
```

Comment:
The hash is never sent to the server so to have proper comment permalinks the id needs to be in the query